pkg/sharky/store.go

```diff
@@ -44,12 +48,11 @@ type Store struct {
 // - maxDataSize - positive integer representing the maximum blob size to be stored
 func New(basedir fs.FS, shardCnt int, maxDataSize int) (*Store, error) {
 	store := &Store{
-		maxDataSize: maxDataSize,
-		writes:      make(chan write),
-		shards:      make([]*shard, shardCnt),
-		wg:          &sync.WaitGroup{},
-		quit:        make(chan struct{}),
-		metrics:     newMetrics(),
+		maxDataSize:     maxDataSize,
+		shards:          make([]*shard, shardCnt),
+		quit:            make(chan struct{}),
+		metrics:         newMetrics(),
+		availableShards: make(chan availableShard),
 	}
 	for i := range store.shards {
 		s, err := store.create(uint8(i), maxDataSize, basedir)
```

Comment:
nit: `maxDataSize` could be accessed in ` store.create(..)`  via property `s.maxDataSize`